src/test/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackServiceTest.java

```diff
@@ -243,6 +261,25 @@ public void shouldGetBodyInGeneric() throws Exception {
         );
     }
 
+    @Test
+    public void shouldRetryOnError() throws Exception {
+        final String testUrl = "http://localhost:" + wiremockPort + "/test-callback-invaliddd";
+        final CallbackResponse callbackResponse = new CallbackResponse();
+        final CaseDetails caseDetails = new CaseDetails();
+        final CaseEvent caseEvent = new CaseEvent();
+        caseEvent.setId("TEST-EVENT");
+
+        stubFor(post(urlMatching("/test-callback-invaliddd.*")).willReturn(
+            okJson(mapper.writeValueAsString(callbackResponse)).withStatus(500)));
+
+        try {
+            callbackService.send(testUrl, Arrays.asList(3, 5), caseEvent, null, caseDetails, String.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        verify(exactly(3), postRequestedFor(urlMatching("/test-callback-invaliddd.*")));
```

Comment:
we are not actually testing we get the expected backoff time. I know it's a bit tricky to test something like that and perhaps it'll result in some messy test code, but given this is a very important feature I wonder whether we should try.

Reply:
I added duration verification as we're already testing the failures and the time needed for retries are already spent in the above test.